src/fondue/Fondue.js

```diff
@@ -209,37 +226,52 @@ export default class Fondue {
 	}
 
 	// Gets all information about the font features.
+	// TODO: if feature has a UI Name ID, return its name
+	//       https://github.com/Pomax/Font.js/issues/73
 	// Usage:
 	//   fondue.features
 	get features() {
-		const featureResult = {};
-		const result = this._raw("GSUB");
-		const featuresRaw = result;
+		const features = {
+			fixed: {},
+			optional: {},
+		};
 
-		if (result) {
-			const features = result.featureList.featureRecords.map(
-				(record) => record.featureTag
-			);
-			const scripts = result.scriptList.scriptRecords.map(
-				(record) => record.scriptTag
-			);
-			// features = features.filter((feature, index) => features.indexOf(feature) === index); // remove doubles
-			features.forEach((feature, index) => {
-				if (!featureResult[feature]) {
-					featureResult[feature] = {
-						...layoutFeature[feature],
-						scripts: {},
-					};
-				}
-				featureResult[feature].scripts[
-					scripts[index % scripts.length]
-				] = {
-					id: scripts[index % scripts.length],
-					_raw: result.getLookups(featuresRaw.getFeature(index)), // .map(lookup => lookup.getSubTables())
-				};
-			});
-		}
-		return featureResult;
+		const getFeats = (result) => {
+			if (result) {
+				// This loops over all features, regardless of script/lang
+				// So it's fine when used to create a list of "all" features
+				// but not when you want to display *which* features belong
+				// to a specific script/language. For now, this is okay.
+				const rawFeatures = result.featureList.featureRecords.map(
+					(record) => record.featureTag
+				);
+
+				rawFeatures.forEach((feature) => {
+					// Translate e.g. cv64 to cv## so it can be found
+					// in the featureMapping
+					let featureIndex = feature;
+					const featureInitial = featureIndex.substring(0, 2);
+					if (featureInitial == "ss" || featureInitial == "cv") {
+						featureIndex = `${featureInitial}##`;
+					}
+					// Map fixed and on/off features to their own set
+					if (!features[feature]) {
+						if (featureMapping[featureIndex].state === "fixed") {
+							features["fixed"][feature] =
+								featureMapping[featureIndex];
+						} else {
+							features["optional"][feature] =
+								featureMapping[featureIndex];
+						}
+					}
+				});
+			}
+		};
+
+		getFeats(this._raw("GSUB"));
+		getFeats(this._raw("GPOS"));
```

Comment:
Hmm this code really makes it seem like it's doing nothing, but it's modifying a variable in scope `const features`.
Could you change this so it's not mutating this variable?

It seems that the code doesn't have to distinguish between GSUB and GPOS, so perhaps you could something like this:

```js
const getRawFeatures = table => tabe.featureList.featureRecords.map(
    (record) => record.featureTag
);

const rawFeatures = [
   ...getRawFeatures(this._raw("GSUB")),
   ...getRawFeatures(this._raw("GPOS")),
];

return rawFeatures.reduce((features, rawFeature) => {
  // mapping
  features["fixed"][rawFeature] = ...;
}, {});
```